server/game/gameSystems/DiscardCardSystem.ts

```diff
@@ -0,0 +1,47 @@
+import type { AbilityContext } from '../core/ability/AbilityContext';
+import { Card } from '../core/card/Card';
+import { EventName, Location } from '../core/Constants';
+import { GameEvent } from '../core/event/GameEvent';
+import { CardTargetSystem, ICardTargetSystemProperties } from '../core/gameSystem/CardTargetSystem';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-object-type
+export interface IDiscardProperties extends ICardTargetSystemProperties {}
+
+export class DiscardCardSystem<TContext extends AbilityContext = AbilityContext> extends CardTargetSystem<TContext, IDiscardProperties> {
+    public override readonly name = 'discardCard';
+    public override readonly eventName = EventName.OnCardsDiscarded;
```

Comment:
Since all forms of discard will probably use this same event type, we should add some additional metadata to the event in case there are ever abilities that care where the card is discarded from. We can add additional properties to the event by overriding `addPropertiesToEvent` (need to be sure to call the superclass method). The we can just do like `event.discardedFromLocation = ...`